microsoft/testsuites/ltp/ltp.py

```diff
@@ -0,0 +1,357 @@
+            self.node.os.install_packages(
+                [
+                    "ntp",
+                    "git-core",
+                    "db48-utils",
+                    "libaio-devel",
+                    "libattr1",
+                    "libcap-progs",
+                    "libdb-4_8",
+                    "perl-BerkeleyDB",
+                ]
+            )
+        elif isinstance(self.node.os, CBLMariner):
+            self.node.os.install_packages(
+                [
+                    "kernel-headers",
+                    "binutils",
+                    "glibc-devel",
+                    "zlib-devel",
+                ]
+            )
+        else:
+            raise LisaException(f"{self.node.os} is not supported")
+
+        # Some CPU time is assigned to set real-time scheduler and it affects
+        # all cgroup test cases. The values for rt_period_us(1000000us or 1s)
+        # and rt_runtime_us (950000us or 0.95s). This gives 0.05s to be used
+        # by non-RT tasks.
+        if self.node.shell.exists(
+            PurePosixPath("/sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us")
+        ):
+            runtime_us = self.node.tools[Cat].read(
+                "/sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us",
+                force_run=True,
+                sudo=True,
+            )
+            runtime_us_int = int(runtime_us)
+            if runtime_us_int == 0:
+                self.node.tools[Echo].write_to_file(
+                    "1000000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/cpu.rt_period_us"),
+                    sudo=True,
+                )
+                self.node.tools[Echo].write_to_file(
+                    "950000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/cpu.rt_runtime_us"),
+                    sudo=True,
+                )
+                self.node.tools[Echo].write_to_file(
+                    "1000000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/user.slice/cpu.rt_period_us"),
+                    sudo=True,
+                )
+                self.node.tools[Echo].write_to_file(
+                    "950000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us"),
+                    sudo=True,
+                )
+
+        # Fix hung_task_timeout_secs and blocked for more than 120 seconds problem
+        sysctl = self.node.tools[Sysctl]
+        sysctl.write("vm.dirty_ratio", "10")
+        sysctl.write("vm.dirty_background_ratio", "5")
+        sysctl.run("-p")
+
+        # define regular stable releases in order to avoid unstable builds
+        # https://github.com/linux-test-project/ltp/tags
+        # 'ltp_version_git_tag' is passed from Test Definition in
+        # .\XML\TestCases\CommunityTests.xml. 'ltp_version_git_tag' default
+        # value is defined in .\XML\Other\ReplaceableTestParameters.xml. You can
+        # run the ltp test with any tag using LISAv2's Custom Parameters feature.
```

Comment:
this is for lisav2, I don't think it is useful for lisav3?

Reply:
fixed